File: shopGunSdkDemo/src/main/java/com/shopgun/android/sdk/demo/PagedPublicationActivity.java
package com.shopgun.android.sdk.demo;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.widget.Toast;

import com.shopgun.android.sdk.demo.base.BaseActivity;
import com.tjek.sdk.api.models.PublicationHotspotV2;
import com.tjek.sdk.api.models.PublicationV2;
import com.tjek.sdk.publicationviewer.paged.IntroConfiguration;
import com.tjek.sdk.publicationviewer.paged.OnHotspotTapListener;
import com.tjek.sdk.publicationviewer.paged.OutroConfiguration;
import com.tjek.sdk.publicationviewer.paged.PagedPublicationConfiguration;
import com.tjek.sdk.publicationviewer.paged.PagedPublicationFragment;

import java.util.List;
import java.util.Locale;

public class PagedPublicationActivity extends BaseActivity {

    public static final String TAG = PagedPublicationActivity.class.getSimpleName();

    public static final String FRAGMENT_TAG = "publication-fragment";

    private static final String KEY_PUB = "PUBLICATION";

    private PagedPublicationFragment mPagedPublicationFragment;

    public static void start(Context context, PublicationV2 publication) {
        Intent i = new Intent(context, PagedPublicationActivity.class);
        i.putExtra(KEY_PUB, publication);
        context.startActivity(i);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pagedpublicationkit);

        mPagedPublicationFragment = (PagedPublicationFragment)
                getSupportFragmentManager().findFragmentByTag(FRAGMENT_TAG);
        if (mPagedPublicationFragment == null) {
            PublicationV2 publication = getIntent().getExtras().getParcelable(KEY_PUB);
            mPagedPublicationFragment = PagedPublicationFragment.Companion.newInstance(
                    publication,
                    new PagedPublicationConfiguration(
                            true,
                            true,
                            null,
                            new OutroConfig()
                    ),
                    0);
            getSupportFragmentManager()
                    .beginTransaction()
                    .add(R.id.pagedPublication, mPagedPublicationFragment, FRAGMENT_TAG)
                    .commit();
        }

        addPagedPublicationListeners();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        mPagedPublicationFragment.clearAdapter();
        super.onSaveInstanceState(outState);
    }

    private void addPagedPublicationListeners() {

//        mPagedPublicationFragment.addOnPageChangeListener(new VersoFragment.OnPageChangeListener() {
//            @Override
//            public void onPagesScrolled(int currentPosition, int[] currentPages, int previousPosition, int[] previousPages) {
//                L.d(TAG, String.format(Locale.US, "onPagesChanged[ currentPosition:%s, currentPages:%s, previousPosition:%s, previousPages:%s ]"
//                        , currentPosition, TextUtils.join(",", currentPages), previousPosition, TextUtils.join(",", previousPages)));
//                String name = getConfig().getCatalog().getBranding().getName();
//                String spread = String.format(" spread[ %s / %s ]", currentPosition+1, getConfig().getSpreadCount());
//
//                if (!getConfig().hasIntro()) {
//                    for (int i = 0; i < currentPages.length; i++) {
//                        currentPages[i] = currentPages[i] + 1;
//                    }
//                }
//                boolean isIntroOutro = (getConfig().hasIntro() && currentPosition == 0) ||
//                        (getConfig().hasOutro() && currentPosition == getConfig().getSpreadCount()-1);
//                String pages = isIntroOutro ? "" : String.format(" page[ %s / %s ]", TextUtils.join("-", currentPages), getConfig().getPublicationPageCount());
//                setTitle(name + spread + pages);
//            }
//
//            @Override
//            public void onPagesChanged(int currentPosition, int[] currentPages, int previousPosition, int[] previousPages) {
//            }
//
//            @Override
//            public void onVisiblePageIndexesChanged(int[] pages, int[] added, int[] removed) {
//
//            }
//        });
//
        mPagedPublicationFragment.setOnHotspotTapListener(new OnHotspotTapListener() {
            @Override
            public void onHotspotTap(@NonNull List<PublicationHotspotV2> hotspots) {
                StringBuilder sb = new StringBuilder();
                for (PublicationHotspotV2 h : hotspots) {
                    if (sb.length() > 0) {
                        sb.append("\n");
                    }
                    if (h.getOffer() != null) {
                        sb.append(h.getOffer().getHeading());
                    }
                }
                Toast.makeText(PagedPublicationActivity.this, sb.toString(), Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onHotspotLongTap(@NonNull List<PublicationHotspotV2> hotspots) {

            }
        });

    }

}

File: shopGunSdkDemo/src/main/java/com/shopgun/android/sdk/demo/PagedPublicationActivity.kt
package com.shopgun.android.sdk.demo

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import android.widget.Toast
import com.shopgun.android.sdk.demo.base.BaseActivity
import com.tjek.sdk.api.models.PublicationHotspotV2
import com.tjek.sdk.api.models.PublicationPageV2
import com.tjek.sdk.api.models.PublicationV2
import com.tjek.sdk.api.remote.ErrorType
import com.tjek.sdk.publicationviewer.paged.*
import com.tjek.sdk.publicationviewer.paged.PagedPublicationFragment.Companion.newInstance

class PagedPublicationActivity : BaseActivity() {

    private var mPagedPublicationFragment: PagedPublicationFragment? = null
    private lateinit var pageTV: TextView

    private var publicationConfiguration = PagedPublicationConfiguration(
        outroConfiguration = OutroConfig()
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_paged_publication_layout)

        pageTV = findViewById(R.id.pageTV)

        mPagedPublicationFragment =
            supportFragmentManager.findFragmentByTag(FRAGMENT_TAG) as PagedPublicationFragment?

        if (mPagedPublicationFragment == null) {
            val publication: PublicationV2 = intent.extras!!.getParcelable(KEY_PUB)!!
            mPagedPublicationFragment = newInstance(
                publication,
                publicationConfiguration
            )
            supportFragmentManager
                .beginTransaction()
                .add(R.id.pagedPublication, mPagedPublicationFragment!!, FRAGMENT_TAG)
                .commit()

            // Note: the initial count needs to be set up here, you won't receive any event (that is triggered when the page changes).
            // It's up to you pass in the correct initial page number
            pageTV.text = "${publicationConfiguration.initialPageNumber + 1} / ${publication.pageCount}"
        }
        addPagedPublicationListeners()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        mPagedPublicationFragment?.clearAdapter()
        super.onSaveInstanceState(outState)
    }

    private fun addPagedPublicationListeners() {
        mPagedPublicationFragment?.setOnPageChangeListener(object : OnPageNumberChangeListener {
            override fun onPageNumberChange(currentPages: IntArray, totalPages: Int) {
                pageTV.text = "${currentPages.joinToString(" - ")} / $totalPages"
            }
        })
        mPagedPublicationFragment?.setOnHotspotTapListener(object : OnHotspotTapListener {
            override fun onHotspotTap(hotspots: List<PublicationHotspotV2>) {
                val sb = StringBuilder()
                for (h in hotspots) {
                    if (sb.isNotEmpty()) {
                        sb.append("\n")
                    }
                    h.offer?.let {
                        sb.append(it.heading)
                    }
                }
                Toast.makeText(this@PagedPublicationActivity, sb.toString(), Toast.LENGTH_SHORT).show()
            }

            override fun onHotspotLongTap(hotspots: List<PublicationHotspotV2>) {}
        })

        mPagedPublicationFragment?.setOnLoadCompleteListener(object : OnLoadComplete {
            override fun onPublicationLoaded(publication: PublicationV2) {
                Log.d(TAG, "publication loaded ${publication.id}")
            }

            override fun onPagesLoaded(pages: List<PublicationPageV2>) {
                Log.d(TAG, "pages loaded ${pages.size}")
            }

            override fun onPageLoad(page: Int) {
                Log.d(TAG, "page loaded $page")
            }

            override fun onHotspotLoaded(hotspots: List<PublicationHotspotV2>) {
                Log.d(TAG, "hotspot loaded ${hotspots.size}")
            }

            override fun onError(error: ErrorType) {}
        })
    }

    companion object {
        val TAG = PagedPublicationActivity::class.java.simpleName
        const val FRAGMENT_TAG = "publication-fragment"
        private const val KEY_PUB = "PUBLICATION"
        fun start(context: Context, publication: PublicationV2?) {
            val i = Intent(context, PagedPublicationActivity::class.java)
            i.putExtra(KEY_PUB, publication)
            context.startActivity(i)
        }
    }
}